pandas/tools/tile.py

```diff
@@ -81,6 +83,17 @@ def cut(x, bins, right=True, labels=None, retbins=False, precision=3,
     array([1, 1, 1, 1, 1], dtype=int64)
     """
     # NOTE: this binning code is changed a bit from histogram for var(x) == 0
+    # for handling the cut for datetime and timedelta objects
+
+    dtype = None
+    if is_timedelta64_dtype(x):
+        x = x.view(np.int64)
+        dtype = np.timedelta64
+
+    elif is_datetime64_dtype(x):
+        x = x.view(np.int64)
+        dtype = np.datetime64
```

Comment:
make a function in this module like

```
def _coerce_to_type(x):
     dtype = None
     original = x
     x = np.asarray(x)
     if is_timedelta64_dtype(x):
....

    return original, x, dtype
```
Then call this (in the 2 places u r using it).

return the orignal data, coerced x, dtype

then also pass the original data (if you need to reconstruct something in side the functions)